app/views/layouts/_footer.html.erb

```diff
@@ -1,8 +1,24 @@
+<% navigation_links = Rails.cache.fetch("navigation_links", expires_in: 15.minutes) do
+{
+    default_nav_ids: NavigationLink.default_section.ordered.ids,
+    other_nav_ids: NavigationLink.other_section.ordered.ids,
+  }
+end %>
```

Comment:
Just thinking out loud, but... I am always nervous about setting variables in views. Seeing how these ids are fed into the `NavigationLink` query and not used directly in the view, I wonder if it might work to move them into a view object's private methods?

Reply:
The reason we've set them previously in `main_nav` in the view instead of in a controller further up the stack is because the partial is being called from multiple places and then we'd need to declare them in multiple places so we took the trade off. However, this case is a little bit different and perhaps I could move it up the stack. I'll look into it in a follow up.